pandas/tests/extension/base/groupby.py

```diff
@@ -33,6 +33,15 @@ def test_groupby_extension_agg(self, as_index, data_for_grouping):
             expected = expected.reset_index()
             self.assert_frame_equal(result, expected)
 
+    def test_groupby_agg_extension(self, data_for_grouping):
+        # GH#38980 groupby agg on extension type fails for non-numeric types
+        df = pd.DataFrame({"A": [1, 1, 2, 2, 3, 3, 1, 4], "B": data_for_grouping})
+        result = df.groupby("A").agg({"B": "first"}).B.array
+
+        expected = df["B"].iloc[[0, 2, 4, 7]].array
+
+        self.assert_extension_array_equal(result, expected)
```

Comment:
```suggestion
        expected = df["B"].iloc[[0, 2, 4, 7]].array

        result = df.groupby("A").agg({"B": "first"}).B.array
        self.assert_extension_array_equal(result, expected)

        result = df.groupby("A").agg("first").B.array
        self.assert_extension_array_equal(result, expected)

        result = df.groupby("A").first().B.array
        self.assert_extension_array_equal(result, expected)
```

Those different ways apparently take a somewhat different code path, given that the two added ones actually still worked (but so worth testing those different ways)